src/main/java/com/google/devtools/build/lib/bazel/bzlmod/IndexRegistry.java

```diff
@@ -117,18 +118,49 @@ private static class BazelRegistryJson {
     String moduleBasePath;
   }
 
-  /** Represents fields available in {@code source.json} for each version of a module. */
+  /** Represents the type field in {@code source.json} for each version of a module. */
   private static class SourceJson {
     String type = "archive";
+  }
+
+  /** Represents fields in {@code source.json} for each archive-type version of a module. */
+  private static class ArchiveSourceJson {
+    String type = "archive";
     URL url;
     String integrity;
     String stripPrefix;
     Map<String, String> patches;
     int patchStrip;
-    String path;
     String archiveType;
   }
 
+  /** Represents fields in {@code source.json} for each local_path-type version of a module. */
+  private static class LocalPathSourceJson {
+    String type = "local_path";
+    String path;
+  }
+
+  /** Represents fields in {@code source.json} for each git_repository-type version of a module. */
+  private static class GitRepoSourceJson {
+    String type = "git_repository";
+    String remote;
+    String commit;
+    String shallowSince;
+    String tag;
+    String branch;
```

Comment:
which of these fields are actually necessary for your use case?

I'm asking because fields like `branch` and especially `patches` are dangerous for a normal registry like the BCR. Ideally, source.json points you to an immutable repo source. I realize the `local_path` literally contradicts this, but at a cursory glance `git_repository` is completely suitable for use in the BCR as long as we stick to the `commit` attribute only.

Fields like `patches` are dangerous as they are actually of `label` type (i.e. workspace-relative), which cannot possibly make sense in a registry context. Note that the `patches` field in `ArchiveSourceJson` actually correspond to the `remote_patches` attribute in `http_archive`. Similar for `buildFile`.

So I'd be more comfortable if we started with a small set of fields that directly satisfy your use case first.

Reply:
I removed the following for now:
branch
patch.*
build.*

Reply:
I propose to submit the current change as-is and patche support in a later change.